cli/testflinger_cli/auth.py

```
@@ -0,0 +1,105 @@
+# Copyright (C) 2020-2022 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Testflinger CLI Auth module."""
+
+import contextlib
+from functools import cached_property
+from http import HTTPStatus
+from typing import Optional
+
+import jwt
+
+from testflinger_cli import client
+from testflinger_cli.errors import AuthenticationError, AuthorizationError
+
+
+class TestflingerCliAuth:
+    """Class to handle authentication and authorisation for Testflinger CLI."""
+
+    def __init__(self, client_id: str, secret_key: str, tf_client):
+        self.client_id = client_id
+        self.secret_key = secret_key
+        self.client = tf_client
+        _ = self.jwt_token
```

I know you mentioned in the comment, but could you add a comment like "fetch JWT to fail fast in case of authentication error" or something shorter, I don't know